### PR TITLE
fix(helm): To fix un-expected {{end}} in helm template

### LINKED ISSUE #11398: Unexpected {{end}} in helm chart
Just found out about this, while playing around  with current helm in EKS.

```shell script
$ helm lint
==> Linting .
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: parse error at (cilium/charts/hubble-ui/templates/serviceaccount.yaml:10): unexpected {{end}}

Error: 1 chart(s) linted, 1 chart(s) failed
```

### PR DESCRIPTION
**Pull request by sayboras**
The block {{end}} at the end was not expected. This should be
removed. Moreover, I add basic helm lint and test to avoid
such kind of issue in future. This is running as non mandatory
step by github action

Closes #11398

Signed-off-by: Tam Mach <sayboras@yahoo.com>

Fixes: #11398

```release-note
fix(helm): To fix un-expected {{end}} in helm template
```

### CODE REVIEW

File: .github/workflows/helm-check.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Helm Lint and Test Charts
+
+on:
+  pull_request:
+    paths:
+      - 'install/kubernetes/**'
```
Comment (sayboras) [inline .github/workflows/helm-check.yaml:6]: this job only kicks off if there is changes for any files matching this path.

**Comment by nebril** [comment]
test-me-please

File: ct.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# See https://github.com/helm/chart-testing#configuration
```
Comment (nebril) [inline ct.yaml]: Can we move this to `install/kubernetes` to avoid polluting root dir?

**Review by nebril — CHANGES_REQUESTED** [review]
Nice! One small change request, otherwise looks good.

**Review by aanm — APPROVED** [review]

**Review by aanm — APPROVED** [review]
Nothing else besides what Maciek said.

File: ct.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+remote: origin
+chart-dirs:
+  - install/kubernetes
+all: true
+debug: true
+validate-maintainers: false
+check-version-increment: false
+helm-extra-args: --timeout 10m0s
```
Comment (joestringer) [inline ct.yaml]: minor nit, is this file required to be in the root of the repo?

In the interests of minimizing the sheer number of files in this directory, it'd be nice to move it somewhere related to github actions; also my first impression of `ct` in the context of Cilium was not "chart testing", but "conntrack" which I found surprising ;-)

**Review by joestringer — COMMENTED** [review]
Looks good, is there any way to address the minor nit I mentioned below?

**Comment by joestringer** [comment]
I wonder if it's worth also adding a `make -C install/kubernetes` and `git diff` check to ensure that the contributor has regenerated the quick-install yaml and committed the changes?

**Comment by michi-covalent** [comment]
> I wonder if it's worth also adding a `make -C install/kubernetes` and `git diff` check to ensure that the contributor has regenerated the quick-install yaml and committed the changes?

@joestringer that would be wonderful, i didn't even know that i'm supposed to regenerate quick-install.yaml :)

i guess that can be a follow-up PR.

File: ct.yaml (hunk 2 above)
Comment (sayboras) [reply ct.yaml]: agreed, let me send update for this.

File: ct.yaml (hunk 3 above)
Comment (sayboras) [reply ct.yaml]: You are right, it's not required to be in root directory. As suggested by @nebril, I thinking _install/kubernetest_ dir is a good place to put. Let me know if you have other idea.

**Comment by sayboras** [comment]
> > I wonder if it's worth also adding a `make -C install/kubernetes` and `git diff` check to ensure that the contributor has regenerated the quick-install yaml and committed the changes?
> 
> @joestringer that would be wonderful, i didn't even know that i'm supposed to regenerate quick-install.yaml :)
> 
> i guess that can be a follow-up PR.

@joestringer @michi-covalent oh right, I was not aware of such make target, let me send subsequent PR for the same.

File: ct.yaml (hunk 3 above)
Comment (sayboras) [reply ct.yaml]: I also renamed ct.yaml to chart-testing.yaml to avoid above confusion, thanks

File: install/kubernetes/Makefile [hunk 4]
```diff
@@ -17,9 +17,11 @@ CILIUM_CHART_REGEX := '\([vV]ersion:\) '$(VERSION_REGEX)
 CILIUM_TAG_REGEX := '\(tag:\) \(v'$(VERSION_REGEX)'\|latest\)'
 CILIUM_PULLPOLICY_REGEX := '\(pullPolicy:\) .*'
 
+.DEFAULT_GOAL := $(QUICK_INSTALL)
+
```
Comment (joestringer) [inline install/kubernetes/Makefile]: Is there a particular reason / concern you have with not having `all` as the default target?

I'm thinking that once we do backports for older branches where we may change the `/VERSION` file, keeping the `update-versions` step will become important to ensure we properly update the quick-install YAML with the correct cilium-agent versions as well.

Comment (sayboras) [reply install/kubernetes/Makefile]: There is no particular reason, it's just that I am not fully aware of cilium's releasing process for helm charts, so didn't want to either do any version check or update helm version. However, I just check the logic for _update-versions_, it just takes _/VERSION_ value and update helm files, so agree to run it as part of github action.

Hence, I will remove above default target, as it was introduced to avoid calling $(QUICK_INSTALL) path. I will just call `make all` in github action.

**Review by aanm — CHANGES_REQUESTED** [review]
@sayboras can you squash all commits together?

**Comment by sayboras** [comment]
> @sayboras can you squash all commits together?

@aanm Done. I rebased with master as well.

**Comment by aanm** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Review by nebril — APPROVED** [review]

**Comment by nebril** [comment]
retest-ginkgo

**Comment by nebril** [comment]
test-with-kernel

**Comment by nebril** [comment]
restart-ginkgo

**Comment by joestringer** [comment]
Hit what looks like a variation on #11313:
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/19615/testReport/junit/Suite-k8s-1/11/K8sServicesTest_Checks_service_across_nodes_Tests_NodePort_BPF_Tests_with_direct_routing_Tests_NodePort/

Pod "testclient-9stfz" can not connect to service "tftp://[::ffff:192.168.36.11]:30438/hello" (failed in request 5/10)

**Comment by joestringer** [comment]
restart-ginkgo

**Comment by sayboras** [comment]
@joestringer seems like the build is successful now, let me know if there is anything else to be addressed :)

**Review by tgraf — APPROVED** [review]
